src/abstract-interpretation/domains/interval-domain.ts

```diff
@@ -139,47 +158,69 @@ implements AbstractDomain<number, IntervalValue, IntervalTop, IntervalBottom, Va
 		return IntervalDomain.abstract(concrete);
 	}
 
+	public satisfies(value: number): Ternary {
+		if(this.isValue() && this.value[0] <= value && value <= this.value[1]) {
+			return this.value[0] === this.value[1] ? Ternary.Always : Ternary.Maybe;
+		}
+		return Ternary.Never;
+	}
+
+	public satisfiesLeq(value: number): Ternary {
+		if(this.isValue() && value <= this.value[1]) {
+			return value <= this.value[0] ? Ternary.Always : Ternary.Maybe;
+		}
+		return Ternary.Never;
+	}
+
 	/**
 	 * Adds another abstract value to the current abstract value by adding the two lower and upper bounds, respectively.
 	 */
-	public add(other: IntervalDomain): IntervalDomain {
-		if(this.value === Bottom || other.value === Bottom) {
+	public add(other: IntervalDomain | IntervalLift): IntervalDomain {
```

Comment:
I accept that we follow a more lisa like implementation of language operators, _but_ what is your opinion on joining this with our function database as i tried to in 

https://github.com/flowr-analysis/flowr/blob/f4e3edcca6d1fe4dcc289d0bc0ba22f75c60bbc2/src/dataflow/eval/values/intervals/interval-binary.ts#L48-L74